patroni/api.py

```diff
@@ -180,6 +180,76 @@ def do_GET_config(self):
         else:
             self.send_error(502)
 
+    def do_GET_metrics(self):
+        postgres = self.get_postgresql_status(True)
+        metrics = []
+
+        metrics.append("# HELP patroni_running Value is 1 if Patroni/Postgres is up, 0 otherwise.")
+        metrics.append("# TYPE patroni_running gauge")
+        if postgres['state'] == 'running':
+            metrics.append("patroni_running 1")
+        else:
+            metrics.append("patroni_running 0")
+
+        try:
+            postmaster_epoch_start_time = int(
+                datetime.datetime.strptime(
+                    postgres['postmaster_start_time'], "%Y-%m-%d %H:%M:%S.%f %Z"
+                ).strftime('%s')
+            )
+            metrics.append("# HELP patroni_postmaster_start_time Epoch seconds since Patroni/Postgres started.")
+            metrics.append("# TYPE patroni_postmaster_start_time counter")
+            metrics.append("patroni_postmaster_start_time {0}".format(postmaster_epoch_start_time))
+        except ValueError:
+            pass
+
+        if postgres['role'] == 'master':
+            metrics.append("# HELP patroni_master Value is 1 if this node is the leader, 0 otherwise.")
+            metrics.append("# TYPE patroni_master gauge")
+            metrics.append("patroni_master 1")
+            metrics.append("# HELP patroni_xlog_location Current location of the Postgres transaction log, 0 if this node is not the leader.")  # noqa: E501
+            metrics.append("# TYPE patroni_xlog_location counter")
+            metrics.append("patroni_xlog_location {0}".format(postgres['xlog']['location']))
+        else:
+            metrics.append("# HELP patroni_master Value is 1 if this node is the leader, 0 otherwise.")
+            metrics.append("# TYPE patroni_master gauge")
+            metrics.append("patroni_master 0")
+            metrics.append("# HELP patroni_xlog_location Current location of the Postgres transaction log, 0 if this node is not the leader.")  # noqa: E501
+            metrics.append("# TYPE patroni_xlog_location counter")
+            metrics.append("patroni_xlog_location 0")
```

Comment:
It pretty much 100% duplicates https://github.com/zalando/patroni/pull/1848/files#diff-13329cd3342d1757c2c9b4dd7a26c0170f4deee5d32eff5c2fd719cc9ff9375bR207-R212
What about using something like following:

```python
        metrics.append("# HELP patroni_master Value is 1 if this node is the leader, 0 otherwise.")
        metrics.append("# TYPE patroni_master gauge")
        metrics.append("patroni_master {0}".format(int(postgres['role'] == 'master')))
        metrics.append("# HELP patroni_xlog_location Current location of the PostgreSQL"
                       " transaction log, 0 if this node is not the leader.")  
        metrics.append("# TYPE patroni_xlog_location counter")
        metrics.append("patroni_xlog_location {0}".format(postgres.get('xlog', {}).get('location', 0)))
```